src/app/api/rooms/[slug]/route.js

```diff
@@ -0,0 +1,169 @@
+import { createServerComponentClient } from "@supabase/auth-helpers-nextjs";
+import { cookies } from "next/headers";
+import { NextResponse } from "next/server";
+
+import prisma from "../../../../../lib/db";
+import { checkMembership } from "../../_utils";
+
+export const dynamic = "force-dynamic";
+
+/** 
+ * @swagger
+ * /api/rooms/[slug]:
+ *   post:
+ *     summary: Add user to existing room
+ *     description: Add another user to an existing room
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               roomId:
+ *                 type: string
+ *                 description: The ID of the room
+ *               profileId:
+ *                 type: string
+ *                 description: The ID of the profile to add
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 message:
+ *                   type: string
+ *       400:
+ *         description: Bad Request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Error'
+ *       500:
+ *         description: Internal Server Error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/Error'
+ *
+ * components:
+ *  schemas:
+ *    Room:
+ *      type: object
+ *      properties:
+ *        id:
+ *          type: string
+ *        name:
+ *          type: string
+ *        metadata:
+ *          type: object
+ *        createdAt:
+ *          type: string
+ *        updatedAt:
+ *          type: string
+ *        deletedAt:
+ *          type: string
+ *        buildingId:
+ *          type: string
+ *        members:
+ *          type: array
+ *          items:
+ *            $ref: '#/components/schemas/RoomMember'
```

Comment:
should be Profile not RoomMember, I got confused with naming - user profile room member 😅 will fix in a future PR